### PR TITLE
fix: segmentSelect issue + segmentation modified event

### PR DESCRIPTION
**Pull request by lscoder**
### Context

- `segmentSelect` example was not working as intended
- `SEGMENTATION_DATA_MODIFIED` was not being triggered for `PlanarFreehandContourSegmentationTool`

### Changes & Results

- changed how `contourSegmentationCompletedListener` finds a viewport with PlanarFreehandContourSegmentationTool tool enabled
- added an event listener to run when an annotation is deleted to update the annotationUIDsMap that contains contour UIDs in each segmentation
- added the contour UID of each annotation created after appending/removing annotations

### CODE REVIEW

File: packages/tools/src/utilities/getViewportsForAnnotation.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * Finds a all matching viewports in terms of the orientation of the annotation data
+ * and the frame of reference. This doesn't mean the annotation IS being displayed
+ * on these viewports, just that it could be by navigating the slice, and/or pan/zoom,
+ * without changing the orientation.
+ *
+ * @param annotation - Annotation to find the viewports that it could display in
+ * @returns All viewports to display in
+ */
+export default function getViewportsForAnnotation(annotation: Annotation) {
```
Comment (sedghi) [inline packages/tools/src/utilities/getViewportsForAnnotation.ts:15]: we have this already 

packages/tools/src/utilities/getViewportForAnnotation.ts

**Review by sedghi — COMMENTED** [review]
see my comment thanks

File: packages/tools/src/utilities/getViewportsForAnnotation.ts (hunk 1 above)
Comment (lscoder) [reply packages/tools/src/utilities/getViewportsForAnnotation.ts:15]: ViewportS (plural)
Later it looks for the viewport that is in a toolGroup with freehand tool

Comment (lscoder) [reply packages/tools/src/utilities/getViewportsForAnnotation.ts:15]: Pushed a change to make `getViewportForAnnotation` call `getViewportsForAnnotation` removing duplicated code

**Review by sedghi — APPROVED** [review]

**Pull request merged**